app/models/comfy/cms/page.rb

```diff
@@ -3,14 +3,28 @@
 class Comfy::Cms::Page < ActiveRecord::Base
   scope :most_popular, -> (number_of_items) { unscoped.order('page_views desc').limit(number_of_items) }
 
-  def self.all_english_articles
+  scope :with_tags, -> (tags) { joins(:taggings).where('tag_id IN (?)', tags.map(&:id)) }
+
+  scope :remove_self_from_results, -> (page) { where.not(id: page.id) }
+
+  scope :remove_alternate_language_page_when_welsh, (lambda do |page|
+    where.not(id: page.mirrors.first.try(:id))
+  end)
+
+  scope :sort_by_tag_similarity, (lambda do |keywords|
+    with_tags(keywords)
+      .group('comfy_cms_pages.id')
+      .order('COUNT(*) DESC, comfy_cms_pages.page_views DESC')
```

Comment:
It might be good to comment this to explain what's going on. If I see `COUNT(*)` I think 'number of records', but its meaning is defined by the GROUP BY here so it'd be nice to signal that.